No error reported for overload ambiguity with extension methods with identical signatures and `params` arrays

**Version Used**: 

main: 7ecf61f6d02b5605c97da1408e0440548f6bf863

**Steps to Reproduce**:

Compile the following:
```csharp
using System.Collections;
using System.Collections.Generic;

class MyCollection<T> : IEnumerable<T>
{
    IEnumerator<T> IEnumerable<T>.GetEnumerator() => throw null;
    IEnumerator IEnumerable.GetEnumerator() => throw null;
}

static class ExtensionsA
{
    public static void Add<T>(this MyCollection<T> collection, params string[] args) { }
}

static class ExtensionsB
{
    public static void Add<T>(this MyCollection<T> collection, params string[] args) { }
}

class Program
{
    static void Main()
    {
        var x = new MyCollection<object>();
        x.Add("");
        
        var y = new MyCollection<object> { "" };
    }
}
```

**Expected Behavior**:

The following errors are expected, as reported by 4.10.0-3.24161.9 (b0be4981).
```
(25,11): error CS0121: The call is ambiguous between the following methods or properties:
    'ExtensionsA.Add<T>(MyCollection<T>, params string[])' and
    'ExtensionsB.Add<T>(MyCollection<T>, params string[])'
(27,44): error CS0121: The call is ambiguous between the following methods or properties:
    'ExtensionsA.Add<T>(MyCollection<T>, params string[])' and
    'ExtensionsB.Add<T>(MyCollection<T>, params string[])'
```

**Actual Behavior**:

No errors.

--- a/src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs
+++ b/src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs
@@ -2332,13 +2332,16 @@ outerDefault:
                     TypeSymbol t1 = m1LeastOverriddenParameters[^1].Type;
                     TypeSymbol t2 = m2LeastOverriddenParameters[^1].Type;
 
-                    if (IsBetterParamsCollectionType(t1, t2, ref useSiteInfo))
+                    if (!Conversions.HasIdentityConversion(t1, t2))
                     {
-                        return BetterResult.Left;
-                    }
-                    if (IsBetterParamsCollectionType(t2, t1, ref useSiteInfo))
-                    {
-                        return BetterResult.Right;
+                        if (IsBetterParamsCollectionType(t1, t2, ref useSiteInfo))
+                        {
+                            return BetterResult.Left;
+                        }
+                        if (IsBetterParamsCollectionType(t2, t1, ref useSiteInfo))
+                        {
+                            return BetterResult.Right;
+                        }
                     }
                 }
             }
@@ -2778,6 +2781,7 @@ outerDefault:
             TypeSymbol t2, CollectionExpressionTypeKind kind2, TypeSymbol elementType2,
             ref CompoundUseSiteInfo<AssemblySymbol> useSiteInfo)
         {
+            Debug.Assert(!Conversions.HasIdentityConversion(t1, t2));
 
             // - T1 is System.ReadOnlySpan<E1>, and T2 is System.Span<E2>, and an implicit conversion exists from E1 to E2
             if (kind1 is CollectionExpressionTypeKind.ReadOnlySpan &&

--- a/src/Compilers/CSharp/Test/Emit2/Semantics/ParamsCollectionTests.cs
+++ b/src/Compilers/CSharp/Test/Emit2/Semantics/ParamsCollectionTests.cs
@@ -5724,6 +5724,52 @@ class Program
         }
 
         [Fact]
+        [WorkItem("https://github.com/dotnet/roslyn/issues/72696")]
+        public void BetterOverload_04_Ambiguity()
+        {
+            var src = """
+using System.Collections;
+using System.Collections.Generic;
+
+class MyCollection<T> : IEnumerable<T>
+{
+    IEnumerator<T> IEnumerable<T>.GetEnumerator() => throw null;
+    IEnumerator IEnumerable.GetEnumerator() => throw null;
+}
+
+static class ExtensionsA
+{
+    public static void Add<T>(this MyCollection<T> collection, params string[] args) { }
+}
+
+static class ExtensionsB
+{
+    public static void Add<T>(this MyCollection<T> collection, params string[] args) { }
+}
+
+class Program
+{
+    static void Main()
+    {
+        var x = new MyCollection<object>();
+        x.Add("");
+
+        var y = new MyCollection<object> { "" };
+    }
+}
+""";
+            var comp = CreateCompilation(src, targetFramework: TargetFramework.Net80, options: TestOptions.ReleaseExe);
+            comp.VerifyDiagnostics(
+                // (25,11): error CS0121: The call is ambiguous between the following methods or properties: 'ExtensionsA.Add<T>(MyCollection<T>, params string[])' and 'ExtensionsB.Add<T>(MyCollection<T>, params string[])'
+                //         x.Add("");
+                Diagnostic(ErrorCode.ERR_AmbigCall, "Add").WithArguments("ExtensionsA.Add<T>(MyCollection<T>, params string[])", "ExtensionsB.Add<T>(MyCollection<T>, params string[])").WithLocation(25, 11),
+                // (27,44): error CS0121: The call is ambiguous between the following methods or properties: 'ExtensionsA.Add<T>(MyCollection<T>, params string[])' and 'ExtensionsB.Add<T>(MyCollection<T>, params string[])'
+                //         var y = new MyCollection<object> { "" };
+                Diagnostic(ErrorCode.ERR_AmbigCall, @"""""").WithArguments("ExtensionsA.Add<T>(MyCollection<T>, params string[])", "ExtensionsB.Add<T>(MyCollection<T>, params string[])").WithLocation(27, 44)
+                );
+        }
+
+        [Fact]
         public void GenericInference()
         {
             var src = """

